### PR TITLE
Add 2 new test "reorder from order history list" and "reorder from order detail"

### PR DESCRIPTION
**Pull request by SD1982**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  1.7.8.x 
| Description?      | Add 2 new test "reorder from order history list" and "reorder from order detail"
| Type?             | refacto 
| Category?         | TE 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | no.
| How to test?      | Can be tested with this 2 commands : `URL_FO='your FO url' TEST_PATH="functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList" npm run specific-test` or `URL_FO='your FO url' TEST_PATH="functional/FO/03_userAccount/01_orderHistory/01_orderDetails/reorderFromOrderDetail" npm run specific-test`
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24364)

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+const orderConfirmationPage = require('@pages/FO/checkout/orderConfirmation');
+const foOrderHistoryPage = require('@pages/FO/myAccount/orderHistory');
+const orderDetailsPage = require('@pages/FO/myAccount/orderDetails');
+
+// Import datas
+const {DefaultCustomer} = require('@data/demo/customer');
+const CartData = require('@data/FO/cart');
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+// Import test context
+const testContext = require('@utils/testContext');
+
+// context
+const baseContext = 'functional_FO_userAccount_orderHistory_reorderFromOrderList';
+
+let browserContext;
+let page;
+
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    });
+
+    it('should validate Step Delivery and go to Payment Step', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToPaymentStep', baseContext);
+
+      const isStepDeliveryComplete = await checkoutPage.goToPaymentStep(page);
+      await expect(isStepDeliveryComplete, 'Step Address is not complete').to.be.true;
+    });
+
+    it('should Pay by back wire and confirm order', async function () {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
    it('should Pay by bank wire and confirm order', async function () {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      const pageTitle = await orderConfirmationPage.getPageTitle(page);
+      await expect(pageTitle).to.equal(orderConfirmationPage.pageTitle);
+
+      const cardTitle = await orderConfirmationPage.getOrderConfirmationCardTitle(page);
+      await expect(cardTitle).to.contains(orderConfirmationPage.orderConfirmationCardTitle);
+    });
+  });
+
+  describe('Go to order list and proceed reorder', async () => {
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+      const lastOrderRowNumber = await foOrderHistoryPage.getNumberOfOrders(page);
+      await foOrderHistoryPage.clickOnReorderLink(page, lastOrderRowNumber);
+
+      const isCheckoutPage = await checkoutPage.isCheckoutPage(page);
+      await expect(isCheckoutPage, 'Browser is not in checkout Page').to.be.true;
+    });
+
+    it('should validate Step Address and go to Delivery Step', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkDeliveryStepForReorder', baseContext);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js:178]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'checkDeliveryStepForReorder', baseContext);
      
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+    });
+
+    it('should validate Step Delivery and go to Payment Step', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToPaymentStepForReorder', baseContext);
+
+      const isStepDeliveryComplete = await checkoutPage.goToPaymentStep(page);
+      await expect(isStepDeliveryComplete, 'Step Address is not complete').to.be.true;
+    });
+
+    it('should Pay by back wire and confirm order', async function () {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
    it('should Pay by bank wire and confirm order', async function () {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+      const pageTitle = await orderConfirmationPage.getPageTitle(page);
+      await expect(pageTitle).to.equal(orderConfirmationPage.pageTitle);
+
+      const cardTitle = await orderConfirmationPage.getOrderConfirmationCardTitle(page);
+      await expect(cardTitle).to.contains(orderConfirmationPage.orderConfirmationCardTitle);
+    });
+  });
+
+  describe('Go to order detail and check content', async () => {
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js]:
```suggestion
```

File: tests/UI/pages/FO/myAccount/orderHistory.js [hunk 8]
```diff
@@ -32,13 +32,24 @@ class OrderHistory extends FOBasePage {
...
   }
 
+
+  /**
+   *
+   * @param page {Page} Browser tab
+   * @param orderRow {Number} row in orders table
+   * @returns {Promise<void>}
+   */
+  async clickOnReorderLink(page, orderRow= 1){
```
Comment (Progi1984) [inline tests/UI/pages/FO/myAccount/orderHistory.js]:
```suggestion
  async clickOnReorderLink(page, orderRow = 1){
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback but it's approved

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 9]
```diff
@@ -41,6 +54,25 @@ class OrderHistory extends FOBasePage {
     await this.clickAndWaitForNavigation(page, this.submitMessageButton);
     return this.getTextContent(page, this.alertSuccessBlock);
   }
+
+  /**
+   * Retrieve and return product name from order detail page
+   * @param page {Page} Browser tab
+   * @param row {Number} row in orders details table
+   * @returns {Promise<string|TextContent|*>}
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/FO/myAccount/orderDetails.js]:
```suggestion
   * @returns {Promise<string>}
```

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 10]
```diff
@@ -41,6 +54,25 @@ class OrderHistory extends FOBasePage {
...
+   */
+  async getProductName(page, row = 1){
+    return this.getTextContent(page, this.productName(row))
+  }
+
+  /**
+   * Get the poge title from the header H1
+   * @param page {Page} Browser tab
+   * @returns {Promise<string|TextContent|*>}
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/FO/myAccount/orderDetails.js]:
```suggestion
   * @returns {Promise<string>}
```

Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]:
```suggestion
   * @override
   * Get the page title from the main section
   * @param page {Page} Browser tab
   * @returns {Promise<string>}
```

File: tests/UI/pages/FO/myAccount/orderHistory.js (hunk 8 above)
Comment (nesrineabdmouleh) [inline tests/UI/pages/FO/myAccount/orderHistory.js]:
```suggestion
  async clickOnReorderLink(page, orderRow = 1){
```

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 11]
```diff
@@ -41,6 +54,25 @@ class OrderHistory extends FOBasePage {
...
   }
+
+  /**
+   * Retrieve and return product name from order detail page
+   * @param page {Page} Browser tab
+   * @param row {Number} row in orders details table
+   * @returns {Promise<string|TextContent|*>}
+   */
+  async getProductName(page, row = 1){
+    return this.getTextContent(page, this.productName(row))
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]:
```suggestion
   * @returns {Promise<string>}
   */
  getProductName(page, row = 1){
    return this.getTextContent(page, this.productName(row))
```

File: tests/UI/pages/FO/myAccount/orderHistory.js [hunk 12]
```diff
@@ -32,13 +32,24 @@ class OrderHistory extends FOBasePage {
...
   isReorderLinkVisible(page, idOrder = 1) {
     return this.elementVisible(page, this.reorderLink(idOrder), 1000);
   }
 
+
+  /**
+   *
+   * @param page {Page} Browser tab
+   * @param orderRow {Number} row in orders table
+   * @returns {Promise<void>}
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderHistory.js]:
```suggestion
   * Click on reorder link
   * @param page {Page} Browser tab
   * @param orderRow {Number} row in orders table
   * @returns {Promise<void>}
```

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 13]
```diff
@@ -13,6 +13,19 @@ class OrderHistory extends FOBasePage {
     this.productIdSelect = '[name=id_product]';
     this.messageTextarea = '[name=msgText]';
     this.submitMessageButton = '[name=submitMessage]';
+    this.headerTitle = `.page-header h1`
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]:
```suggestion
    this.headerTitle = `.page-header h1`;
```

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 14]
```diff
@@ -59,19 +60,24 @@ class OrderHistory extends FOBasePage {
...
-   * @returns {Promise<string|TextContent|*>}
+   * @returns {Promise<string>}
    */
-  async getPageTitle(page){
-    return this.getTextContent(page, this.headerTitle)
+  async getPageTitle(page) {
+    return this.getTextContent(page, this.headerTitle);
+  }
+
+  async clickOnReorderLink(page) {
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]: Please add a function description here

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/reorderFromOrderDetail.js [hunk 15]
```diff
@@ -0,0 +1,253 @@
...
+// Import datas
+const {DefaultCustomer} = require('@data/demo/customer');
+const CartData = require('@data/FO/cart');
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+// context
+const baseContext = 'functional_FO_userAccount_orderHistory_orderDetail_reorderFromOrderList';
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/reorderFromOrderDetail.js]:
```suggestion
const baseContext = 'functional_FO_userAccount_orderHistory_orderDetail_reorderFromOrderDetail';
```

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/reorderFromOrderDetail.js [hunk 16]
```diff
@@ -0,0 +1,254 @@
...
+// Import datas
+const {DefaultCustomer} = require('@data/demo/customer');
+const CartData = require('@data/FO/cart');
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+// context
+const baseContext = 'functional_FO_userAccount_orderHistory_orderDetail_reorderFromOrderDetail';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/reorderFromOrderDetail.js]:
```suggestion
const baseContext = 'functional_FO_userAccount_orderHistory_orderDetails_reorderFromOrderDetail';
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+// Import datas
+const {DefaultCustomer} = require('@data/demo/customer');
+const CartData = require('@data/FO/cart');
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+// context
+const baseContext = 'functional_FO_userAccount_orderHistory_reorderFromOrderList';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js:28]: Shouldn't be `functional_FO_userAccount_orderHistory_orderDetails_reorderFromOrderList` ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks but good job.

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js (hunk 17 above)
Comment (SD1982) [reply tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js:28]: Hi @Progi1984 thanks for the review !!
No this one in not in the orderDetails directory 
The orderDetails directory is in the orderHistory directory
But for this test the parent directory is the orderHistory directory

Comment (Progi1984) [reply tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/reorderFromOrderList.js:28]: :ok:

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @SD1982 & @boubkerbribri

**Pull request merged**